### PR TITLE
Cleanup Blobstore Repository Metadata Serialization

### PR DESCRIPTION
**Pull request by original-brownbear**
Follow up to #62684 making use of shorter utility for corruption checks.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java [hunk 1]
```diff
@@ -507,39 +508,36 @@ public static BlobStoreIndexShardSnapshot fromXContent(XContentParser parser) th
...
-                        }
-                    } else if (token == XContentParser.Token.START_ARRAY) {
-                        if (PARSE_FILES.match(currentFieldName, parser.getDeprecationHandler())) {
-                            while ((parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                                indexFiles.add(FileInfo.fromXContent(parser));
-                            }
-                        } else {
-                            throw new ElasticsearchParseException("unknown parameter [{}]", currentFieldName);
+                XContentParserUtils.ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, parser);
+                String currentFieldName = parser.currentName();
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java:512]: I'd prefer this to be `final`

File: server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java [hunk 2]
```diff
@@ -507,39 +508,36 @@ public static BlobStoreIndexShardSnapshot fromXContent(XContentParser parser) th
...
+                        while ((parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                            indexFiles.add(FileInfo.fromXContent(parser));
                         }
                     } else {
-                        throw new ElasticsearchParseException("unexpected token  [{}]", token);
+                        throw new ElasticsearchParseException("unknown parameter [{}]", currentFieldName);
                     }
                 } else {
-                    throw new ElasticsearchParseException("unexpected token [{}]", token);
+                    throw new ElasticsearchParseException("unexpected token  [{}]", token);
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java:540]: Do you really prefer with an extra space? :)

**Review by tlrx — APPROVED** [review]
LGTM, thanks for using these methods Armin.  I left two very minor comments.

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Pull request merged**